src/Aspire.Hosting/ApplicationModel/ResourceLoggerService.cs

```diff
@@ -352,6 +352,14 @@ public void Complete()
             _logStreamCts.Cancel();
         }
 
+        public void Reset()
```

Comment:
A comment on this method would be helpful.

It doesn't seem like the race condition within this method is too serious, but you still might consider enclosing the test-and-write within `lock (_lock)`.